client.go

```diff
@@ -64,11 +66,33 @@ func (c *Client) ProvisionResource(ctx context.Context, cbID, resID manifold.ID,
 	p.SetContext(ctx)
 	res, acceptedRes, noContent, err := c.api.Resource.PutResourcesID(p)
 
+	if err != nil {
+		var graftonErr *Error
+		switch e := err.(type) {
+		case *resource.PutResourcesIDBadRequest:
+			graftonErr = NewError(merrors.BadRequestError, ptrToString(e.Payload.Message))
+		case *resource.PutResourcesIDUnauthorized:
+			graftonErr = NewError(merrors.UnauthorizedError, ptrToString(e.Payload.Message))
+		case *resource.PutResourcesIDConflict:
+			graftonErr = NewError(merrors.ConflictError, ptrToString(e.Payload.Message))
+		case *resource.PutResourcesIDInternalServerError:
+			graftonErr = NewError(merrors.InternalServerError, ptrToString(e.Payload.Message))
```

Comment:
why not do something like:
` graftonErr = NewErrWithMsg(merrors.InternalServerError, e.Payload.Message)`
NewErrWithMsg can do both the ptr to string conversion, and if the string is nil, return ErrMissingMsg instead.